.github/workflows/format.yml

```diff
@@ -15,23 +15,14 @@ jobs:
         with:
           python-version: "3.10"
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
```

Comment:
We should sha pin this,
`astral-sh/setup-uv@0c5e2b8115b80b4c7c5ddf6ffdd634974642d182`
https://github.com/astral-sh/setup-uv/releases/tag/v5.4.1

Reply:
I have a separate item to go through and sha pin all actions. Let's land this because it will have frequent conflicts and then fast follow @CarsonRoscoe